service/src/main/java/bio/terra/drshub/services/TrackingService.java

```diff
@@ -0,0 +1,71 @@
+package bio.terra.drshub.services;
+
+import bio.terra.bard.api.BardApi;
+import bio.terra.bard.model.Event;
+import bio.terra.bard.model.EventProperties;
+import bio.terra.common.iam.BearerToken;
+import bio.terra.drshub.config.DrsHubConfig;
+import com.google.common.annotations.VisibleForTesting;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.map.PassiveExpiringMap;
+import org.springframework.http.ResponseEntity;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public class TrackingService {
+  static final String APP_ID = "drshub";
+
+  private final BardApiFactory bardApiFactory;
+  private final boolean trackInMixpanel;
+
+  private final Map<String, String> bearerTokenCache =
+      Collections.synchronizedMap(new PassiveExpiringMap<>(15, TimeUnit.MINUTES));
+
+  public TrackingService(BardApiFactory bardApiFactory, DrsHubConfig config) {
+    this.bardApiFactory = bardApiFactory;
+    this.trackInMixpanel = config.trackInMixPanel();
+  }
+
+  @Async("asyncExecutor")
+  public void logEvent(BearerToken bearerToken, String eventName, Map<String, ?> properties) {
+    var bardApi = bardApiFactory.getApi(bearerToken);
+    // Sync the user profile if needed.
+    syncUser(bardApi, bearerToken);
+    // Log the user event
+    logEvent(bardApi, eventName, properties);
+  }
+
+  /**
+   * Call the syncProfile endpoint in Bard. This ensures that the Terra user is properly associated
+   * with a MixPanel user. Note that this is if mixpanel tracking is enabled.
```

Comment:
I'm not solid here on the behavior when mixpanel tracking isn't enabled: is this method never called, do the internals behave as no-ops, etc?

Reply:
Oh that's branching that lives in Bard.  We always log to the warehouse but we can opt out of _also_ logging to mixpanel (which we want to do in this case in all likelihood to not explode our quota)